docs/kubernetes/user-guides/snapshots.md

```diff
@@ -122,3 +125,150 @@
+           volumeSnapshotContentName: snapcontent-migrated
+     EOF
+     ```
+  1. Create a `VolumeSnapshotContents` pointing to your existing PD
+     snapshot from the first step.
+     ```console
+     kubectl apply -f - <<EOF
+       apiVersion: snapshot.storage.k8s.io/v1beta1
+       kind: VolumeSnapshotContents
+       metadata:
+         name: restored-snapshot-content
+       spec:
+         deletionPolicy: Retain
+         driver: pd.csi.storage.gke.io
+         source:
+           snapshotHandle: projects/$PROJECT_ID/global/snapshots/$SNAPSHOT_ID
+         volumeSnapshotRef:
+           kind: VolumeSnapshot
+           name: restored-snapshot
+           namespace: default
+     EOF
+     ```
+  1. Create a `PersistentVolumeClaim` which will pull from the
+     `VolumeSnapshot`. The `StorageClass` must match what you use to provision
+     PDs; what is below matches the first example in this document.
+     ```console
+     kubectl apply -f - <<EOF
+       kind: PersistentVolumeClaim
+       apiVersion: v1
+       metadata:
+         name: restored-pvc
+       spec:
+         accessModes:
+           - ReadWriteOnce
+         storageClassName: csi-gce-pd
+         resources:
+           requests:
+             storage: 6Gi
+         dataSource:
+           kind: VolumeSnapshot
+           name: restored-snapshot
+           apiGroup: snapshot.storage.k8s.io
+     EOF
+     ```
+  1. Finally, create a pod referring to the `PersistentVolumeClaim`. The PD CSI
+     driver will provision a `PersistentVolume` and populate it from the
+     snapshot. The pod from `examples/kubernetes/snapshot/restored-pod.yaml` is
+     set to use the PVC created above. After `kubectl apply`'ing the pod, run
+     `kubectl exec restored-pod -- ls -l /demo/data/` to confirm that the
+     snapshot has been restored correctly.
+
+#### Troubleshooting
+
+If the `VolumeSnapshot`, `VolumeSnapshotContents` and `PersistentVolumeClaim`
+are not all mutually syncrhonized, the pod will not start. Try `kubectl describe
+volumesnapshotcontent restored-snapshot-content` to see any error messages
+relating to the binding together of the `VolumeSnapshot` and the
+`VolumeSnapshotContents`. Further errors may be found in the snapshot controller
+logs:
+
+```console
+kubectl logs snapshot-controller-0 | tail
+```
+
+Any errors in `kubectl describe pvc restored-pvc` may also shed light on any
+troubles. Sometimes there is quite a wait to dynamically provision the
+`PersistentVolume` for the PVC; the following command will give more
+information.
+
+```console
+kubectl logs -n gce-pd-csi-driver csi-gce-pd-controller-0 -c csi-provisioner | tail
+```
+
+### Tips on Migrating from Alpha Snapshots
+
+The api version has changed between the alpha and beta releases of the CSI
+snapshotter. In many cases, snapshots created with the alpha will not be able to
+be used with a beta driver. In some cases, they can be migrated.
+
+*You may lose all data in your snapshot!*
```

Comment:
Bold and add "Attention" similar to the other notes

Reply:
Done